CMakeLists.txt

```diff
@@ -92,7 +92,7 @@ set(GCC_CXX_WARNING_FLAGS
   "-Wduplicated-cond" "-Wextra-semi" "-Wfloat-equal" "-Wformat-overflow=2"
   "-Wformat-signedness" "-Wformat-truncation=2" "-Wformat=2"
   "-Wimplicit-fallthrough=5" "-Winvalid-pch" "-Wlogical-op" "-Wmismatched-tags"
-  "-Wmissing-declarations" "-Wmissing-include-dirs" "-Wnoexcept"
```

Comment:
@laurynas-biveinis Was it deliberate to remove "-Wmissing-declarations" from the build?

Reply:
Yes: moving code to header together with utility functions caused warnings for those functions that had no other reasonable fix (duplicating a declaration in the same header seemed redundant)